src/server/game/Spells/Auras/SpellAuras.cpp

```diff
@@ -2039,6 +2042,46 @@ void Aura::TriggerProcOnEvent(uint8 procEffectMask, AuraApplication* aurApp, Pro
         Remove();
 }
 
+void Aura::HeartbeatResistance(uint32 diff)
+{
+    m_heartBeatTimer += diff;
+
+    if (m_heartBeatTimer < 950)
+        return;
+
+    Unit* target = GetOwner()->ToUnit();
+    Unit* caster = GetCaster();
+
+    if (!target || !caster)
+        return;
+
+    if (target->GetTypeId() == TYPEID_UNIT && caster->GetTypeId() == TYPEID_PLAYER)
+    {
+        m_heartBeatTimer = 0;
+
+        uint32 auraTimePassed = time(nullptr) - GetApplyTime();
+
+        if (auraTimePassed == (uint32)std::floor((GetMaxDuration() * 0.25f / IN_MILLISECONDS) + 0.5f) ||
+            auraTimePassed == (uint32)std::floor((GetMaxDuration() * 0.50f / IN_MILLISECONDS) + 0.5f) ||
+            auraTimePassed == (uint32)std::floor((GetMaxDuration() * 0.75f / IN_MILLISECONDS) + 0.5f))
+        {
+            TC_LOG_DEBUG("spells", "Aura::HeartbeatResistance: Breaking creature aura. Duration %u, Max %u", auraTimePassed, (GetMaxDuration() / IN_MILLISECONDS));
+
+            SpellSchoolMask schoolMask = m_spellInfo->GetSchoolMask();
+            uint32 resistance = schoolMask != SPELL_SCHOOL_MASK_NORMAL ? target->GetResistance(GetFirstSchoolInMask(schoolMask)) : 0;
+            uint32 breakChance = urand(0, 100);
+            uint32 breakPct = 5 + uint32((resistance / powf(target->getLevel(), 1.441f) * 0.10f) * 100);
+
+            if (breakChance < breakPct)
```

Comment:
ew, no better way of doing a random chance check? (hint, yes)

Reply:
(only skimming this because asked to, somebody else can post what the actual call is, but we definitely have one)

Reply:
`roll_chance_i`